### PR TITLE
kubeadm ports updates

### PR DESCRIPTION
**Pull request by dixudx**
Fixes: kubernetes/kubeadm#770

xref: kubernetes/kubeadm#732, kubernetes/kubeadm#838
xref: #8846

/cc luxas timothysc neolit123

### CODE REVIEW

**Comment by dixudx** [comment]
@seh FYI

File: content/en/docs/tasks/tools/install-kubeadm.md [hunk 1]
```diff
@@ -52,22 +52,20 @@ route, we recommend you add IP route(s) so Kubernetes cluster addresses go via t
...
 
-| Protocol | Direction | Port Range | Purpose                 |
-|----------|-----------|------------|-------------------------|
-| TCP      | Inbound   | 6443*      | Kubernetes API server   |
-| TCP      | Inbound   | 2379-2380  | etcd server client API  |
-| TCP      | Inbound   | 10250      | Kubelet API             |
-| TCP      | Inbound   | 10251      | kube-scheduler          |
-| TCP      | Inbound   | 10252      | kube-controller-manager |
-| TCP      | Inbound   | 10255      | Read-only Kubelet API   |
+| Protocol | Direction | Port Range | Purpose                 | Source Components         |
```
Comment (liztio) [inline content/en/docs/tasks/tools/install-kubeadm.md]: does "source components" here mean "components that need access?" I think that could be a little clearer. "Clients" maybe?

Comment (dixudx) [reply content/en/docs/tasks/tools/install-kubeadm.md]: @liztio "Clients" is better.
@luxas @timothysc Any suggestions? 
Naming is always the hardest part.

Comment (kad) [reply content/en/docs/tasks/tools/install-kubeadm.md]: "used by" ?

File: content/en/docs/tasks/tools/install-kubeadm.md [hunk 2]
```diff
@@ -52,22 +52,20 @@ route, we recommend you add IP route(s) so Kubernetes cluster addresses go via t
...
+|----------|-----------|------------|-------------------------|---------------------------|
+| TCP      | Inbound   | 6443*      | Kubernetes API server   | All                       |
+| TCP      | Inbound   | 2379-2380  | etcd server client API  | kube-apisever, etcd peers |
```
Comment (detiber) [inline content/en/docs/tasks/tools/install-kubeadm.md]: etcd server and peer ports should probably be broken out. While it doesn't really matter when etcd is co-located with the control plane components, if etcd is broken out the firewall requirements are different for each port.

File: content/en/docs/tasks/tools/install-kubeadm.md [hunk 3]
```diff
@@ -52,22 +52,20 @@ route, we recommend you add IP route(s) so Kubernetes cluster addresses go via t
...
 ### Worker node(s)
 
-| Protocol | Direction | Port Range  | Purpose               |
-|----------|-----------|-------------|-----------------------|
-| TCP      | Inbound   | 10250       | Kubelet API           |
-| TCP      | Inbound   | 10255       | Read-only Kubelet API |
-| TCP      | Inbound   | 30000-32767 | NodePort Services**   |
+| Protocol | Direction | Port Range  | Purpose               | Source Components       |
+|----------|-----------|-------------|-----------------------|-------------------------|
+| TCP      | Inbound   | 10250       | Kubelet API           | Self                    |
```
Comment (detiber) [inline content/en/docs/tasks/tools/install-kubeadm.md]: Kubelet api needs to be available to the control plane hosts.

File: content/en/docs/tasks/tools/install-kubeadm.md (hunk 2 above)
Comment (neolit123) [reply content/en/docs/tasks/tools/install-kubeadm.md]: `s/kube-apisever/kube-apiserver`

**Comment by zacharysarah** [comment]
@dixudx 👋 Bump for responses to reviews.

**Comment by Bradamant3** [comment]
/assign

@dixudx is this for 1.11? Ping me in Slack please, thanks.

File: content/en/docs/tasks/tools/install-kubeadm.md (hunk 2 above)
Comment (dixudx) [reply content/en/docs/tasks/tools/install-kubeadm.md]: > While it doesn't really matter when etcd is co-located with the control plane components

Mostly it is true. But kubeadm does provide a option to install etcd together with control plane by etcd manifest. So I prefer to keep it. @luxas WDYT?

File: content/en/docs/tasks/tools/install-kubeadm.md [hunk 4]
```diff
@@ -52,22 +52,20 @@ route, we recommend you add IP route(s) so Kubernetes cluster addresses go via t
...
-| TCP      | Inbound   | 10250      | Kubelet API             |
-| TCP      | Inbound   | 10251      | kube-scheduler          |
-| TCP      | Inbound   | 10252      | kube-controller-manager |
-| TCP      | Inbound   | 10255      | Read-only Kubelet API   |
+| Protocol | Direction | Port Range | Purpose                 | Source Components         |
+|----------|-----------|------------|-------------------------|---------------------------|
+| TCP      | Inbound   | 6443*      | Kubernetes API server   | All                       |
+| TCP      | Inbound   | 2379-2380  | etcd server client API  | kube-apisever, etcd peers |
+| TCP      | Inbound   | 10250      | Kubelet API             | Self                      |
+| TCP      | Inbound   | 10251      | kube-scheduler          | Self                      |
```
[the remark this answers is not in the document]
Comment (dixudx) [reply content/en/docs/tasks/tools/install-kubeadm.md]: @seh  Sorry for the late reply.

Actually here `self` means the port is used to expose its own service locally (127.0.0.1). It is usually consumed by administrators for monitoring and debuging.

**Comment by dixudx** [comment]
@Bradamant3 Sorry for the late reply. IMO We'd better keep it for 1.11, especially for the first commit.

File: content/en/docs/tasks/tools/install-kubeadm.md [hunk 5]
```diff
@@ -52,22 +52,20 @@ route, we recommend you add IP route(s) so Kubernetes cluster addresses go via t
...
-| TCP      | Inbound   | 2379-2380  | etcd server client API  |
-| TCP      | Inbound   | 10250      | Kubelet API             |
-| TCP      | Inbound   | 10251      | kube-scheduler          |
-| TCP      | Inbound   | 10252      | kube-controller-manager |
-| TCP      | Inbound   | 10255      | Read-only Kubelet API   |
+| Protocol | Direction | Port Range | Purpose                 | Source Components         |
+|----------|-----------|------------|-------------------------|---------------------------|
+| TCP      | Inbound   | 6443*      | Kubernetes API server   | All                       |
+| TCP      | Inbound   | 2379-2380  | etcd server client API  | kube-apisever, etcd peers |
+| TCP      | Inbound   | 10250      | Kubelet API             | Self                      |
```
[the remark this answers is not in the document]
Comment (dixudx) [reply content/en/docs/tasks/tools/install-kubeadm.md]: Right.

**Comment by timothysc** [comment]
@Bradamant3 ^ ready to go.

**Pull request merged**